### PR TITLE
build: bump go to 1.22

### PR DESCRIPTION
**Pull request by kruskall**
Bump go version to 1.22

Unblock https://github.com/elastic/apm-aws-lambda/pull/479

### CODE REVIEW

**Review by christos68k — APPROVED** [review]

**Comment by christos68k** [comment]
Are the test failures significant?

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/elastic/apm-aws-lambda
 
-go 1.21
+go 1.22
```
Comment (dmathieu) [inline go.mod:3]:
```suggestion
go 1.22.3
```

File: tools/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/elastic/apm-aws-lambda/tools
 
-go 1.21
+go 1.22
```
Comment (dmathieu) [inline tools/go.mod:3]:
```suggestion
go 1.22.3
```

**Review by dmathieu — COMMENTED** [review]
I think this should make the toolchain retrievable.

**Review by dmathieu — APPROVED** [review]

**Pull request merged**